backend/tournesol/admin.py

```diff
@@ -226,31 +226,40 @@ class PollAdmin(admin.ModelAdmin):
         'name',
         'algorithm',
         'entity_type',
-        'get_n_criterias',
+        'get_n_criteria',
         'get_n_comparisons',
         'get_n_comparisons_per_criteria',
     )
-    list_filter = (
-        'algorithm',
-        'entity_type'
-    )
+    list_filter = ("algorithm", "entity_type")
     inlines = (CriteriasInline,)
 
-    @admin.display(description="# criterias")
-    def get_n_criterias(self, obj):
-        return obj.criterias.count()
+    def get_queryset(self, request):
+        qst = super().get_queryset(request)
+        qst = qst.annotate(_n_criteria=Count("criterias", distinct=True))
+        qst = qst.annotate(_n_comparisons=Count("comparisons", distinct=True))
+        qst = qst.annotate(
+            _n_comparisons_per_criteria=Count(
+                "comparisons__criteria_scores", distinct=True
+            )
+        )
+        return qst
```

Comment:
The multiple `LEFT OUTER JOIN` in the SQL query are actually badly optimized here by Postgres. `COUNT(DISTINCT ...)` seems to be slow in this case, whereas it's supposed to simply count the rows from a single table.

If we really want to use a single query, I think that`Subquery` should be preferred, with something like https://stackoverflow.com/a/65613047/5634396
Although arguably it makes things complicated with little advantage compared to the approach existing on "main", where each poll triggers 3 simple `COUNT()`.

Reply:
That's true, the generated query goes against it primary objective: optimizing / simplifying.

I'll make more test, just by curiosity. If they are conclusive we could close this PR.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/797/commits/e431b7c1db735ad0e7689a96a6483e0ce4620ffc.